### ISSUE TITLE
please do not create two instancesof groupedExternalNodes of the same set of nodes

### ISSUE THREAD

**Issue by haim-kermany** [np-guard/vpc-network-config-analyzer#166]
in d.conn.GroupedConnectivity.GroupedLines
I see that there are two instances of groupedExternalNodes that has the same set of nodes.

**Comment by ShiriMoran** [comment]
This seems like a bug. Can you please refer me to the specific example?

**Comment by haim-kermany** [comment]
acl_testing3

**Comment by haim-kermany** [comment]
also, maybe GroupedConnLine should keep a pointer to the Conn struct, and not keep it as a string

**Comment by ShiriMoran** [comment]
`GroupedConnLine` are created as per the grouping, one `GroupedConnLine` for each grouping item.
Thus, semantically equal  `GroupedConnLine` may reside in source of one line and destination of another (or even the same) line; or in sources of two separate lines that can no be grouped - either since vsis of different subnets are not grouped or since vsis are not grouped with external addressed. 
E.g, in the following example, 3 semantically identical `GroupedConnLine` will be generated for `vsi1a, vsi1b`
```
vsi1a, vsi1b (of subnet1) => vsi2 (of subnet2)
vsi1a, vsi1b (of subnet1) => vsi3 (of subnet3)
vsi1a, vsi1b (of subnet1) => Public Internet (all ranges) : All Connections
```

**Comment by haim-kermany** [comment]
also:
it seems that in a groupedEndpointsElems there are Nodes which does not exist in cConfig.Nodes

**Comment by ShiriMoran** [comment]
solved by PR https://github.com/np-guard/vpc-network-config-analyzer/pull/174

**Comment by ShiriMoran** [comment]
@haim-kermany may I close this issue?

**Comment by haim-kermany** [comment]
Hi
I need this fix also for groupedEndpointsElems of subnets

**Comment by ShiriMoran** [comment]
@haim-kermany In the context of this issue the subnets are handled the same way vsis are. The fix of PR #174 treats subnets and vsis alike.
Hence, if something is not working properly its a bug (the most probably effects both vsis and subnets) that should be analyzed and fixed. If this is the case lets take it offline.

**Comment by ShiriMoran** [comment]
After talking with @haim-kermany 
The fix in #174 assures that vsis and subnets are not duplicated in the grouping.
But more than that is required here. The solution must be extended so that subsets of vsis/subnets are not duplicated - specifically when one is in source and the other in dst.

**Comment by ShiriMoran** [comment]
After taking a closer look, here's the issue: the original fix was perfect. It was ruined by https://github.com/np-guard/vpc-network-config-analyzer/issues/98, since the grouping there is done simultaneously of `src` and `dst`  and thus the fix here did not work.
